TechEmpower/FrameworkBenchmarks
# Use .NET Core 2.0.0-preview1

- move from dotnet-dev-1.0.0-preview2.1-003177 to dotnet-dev-2.0.0-preview1
- convert .NET Core Benchmarks project from project.json to .csproj
  - use latest available versions of all dependencies
- catch up with latest code in Microsoft ASP.NET fork
  - do not grab scoped services when constructing singletons; primarily affects DbContexts
  - compile EF queries
  - WebListener -> HttpSys
  - enable specific requested scenarios e.g. don't support /mvc/plaintext when plaintext requested
  - remove hack and use `response.ContentLength` property; hack no longer a significant optimization
  - remove use of old PlatformAbstractions package
- publish as part of test scenarios and enable MVC Razor Precompilation
- change `DebugInfoPageMiddleware` to return 404 status code
  - prevent apparent success when "benchmarking" incorrect URLs
- add appsettings.json file for Windows runs; use LocalDb by default on Windows

Improve database scenario performance (beyond compiling EF queries)
- enforce Max Auto Prepare for PostgreSQL EF/Dapper
  - merge from https://github.com/roji/FrameworkBenchmarks.git dougbu/update.net.core-autoprepare
  - thanks @roji!
  - The "Max Auto Prepare" connection string parameter is important for PostgreSQL performance when running
    EF/Dapper benchmarks. Added this parameter in appsetings.postgresql.json and a check for its existence
    on benchmark startup.
- `Prepare()` the update command in `RawDb.LoadMultipleUpdatesRows()`
  - may improve performance with Postgres even though the command is executed just once
- remove `NoTransactionSqlServerConnection` hack; leave auto transactions enabled

nits:
- remove unused tool dependencies
- display where `WebHostBuilder` is found during startup; confirms use of publish folder
- slightly improve Windows scripts
- add appsettings.postgresql.json to csproj
- remove obsolete launchSettings.json
- add and correct links in README.md
- remove duplicate property setting from benchmark_config.json
- clean up whitespace
- more `var`

- simplify `services.AddDbContextPool<TContext>()` delegate
